src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -359,46 +363,51 @@ private static string GetNextMethodName(ArrayBuilder<MethodSymbol> builder)
                         }
                     }
 
-                    // Hoisted method parameters (represented as locals in the EE).
-                    if (!_hoistedParameterNames.IsEmpty)
-                    {
-                        int localIndex = 0;
-                        foreach (var local in _localsForBinding)
-                        {
-                            // Since we are showing hoisted method parameters first, the parameters may appear out of order
-                            // in the Locals window if only some of the parameters are hoisted.  This is consistent with the
-                            // behavior of the old EE.
-                            if (_hoistedParameterNames.Contains(local.Name))
-                            {
-                                AppendLocalAndMethod(localBuilder, methodBuilder, local, container, localIndex, GetLocalResultFlags(local));
-                            }
-
-                            localIndex++;
-                        }
-                    }
-
-                    // Method parameters (except those that have been hoisted).
+                    var itemsAdded = new HashSet<string>();
```

Comment:
Please use `PooledHashSet<string>.GetInstance()`, with corresponding `Free()`.

Reply:
Thanks!

---
In reply to: [135910352](https://github.com/dotnet/roslyn/pull/21707#discussion_r135910352) [](ancestors = 135910352)